Source/DafnyDriver/DafnyDriver.cs

```diff
@@ -563,6 +564,25 @@ private record TargetPaths(string Directory, string Filename) {
       var compiler = DafnyOptions.O.Compiler;
 
       var hasMain = Compilers.SinglePassCompiler.HasMain(dafnyProgram, out var mainMethod);
+      if (DafnyOptions.O.RunAllTests) {
```

Comment:
The PR currently seems to work both by generating Dafny code early on and by injecting extra output code in the compiler classes. Why do we need to do both?

Reply:
It has to create the `Method` AST node so that the special-cased logic dealing with the top-level entry point in `DafnyDriver` still works correctly. But I think I'll go ahead with your suggestion of an internal-only AST code for the reflective features so that everything can happen in the resolver instead.